docs/source/background/scientific_software.rst

```diff
@@ -0,0 +1,199 @@
+
+To summarize, while legacy software represents a substantial investment in
+terms of time and money, in many cases existing legacy packages are
+unsustainable. The number of organizations dedicated to developing better
+research software is a direct result of not wanting to repeat the same
+mistakes moving forward.
+
+.. _what_sets_scientific_software_apart:
+
+************************************
+What Sets Scientific Software Apart?
+************************************
+
+As suggested by the intro to to this page, there is an increasing interest in
+developing better more sustainable scientific software. If we are to capitalize
+on these efforts we need to understand what makes developing scientific
+software challenging, and how to avoid repeating the problems of legacy
+software. With regards to why scientific software is unique:
+
+1. Performance
+
+   - Scientific software is among the most computationally expensive software
+     in the world. High computational complexity of many algorithms means that
+     even a small degradation in performance can result in a simulation
+     becoming intractable.
+   - Often requires high-performance computing
+   - Performance is heavily tied to hardware; as hardware evolves so will the
+     scientific software.
+
+#. Scientific motivation
+
+   - Software is typically seen as a means to an end and usually developed by
+     the scientists themselves.
+   - Benefit to cost ratio of dependencies must be large, i.e., dependencies
+     usually only considered if they save a lot of time, or are very performant.
+   - Most scientists prefer to do as little software development as possible.
+
+#. Dynamic nature of scientific research
+
+   - Scientific research is by its nature highly uncertain. Promising
+     avenues may not pan out. Funding sources dry up. New hot topics emerge.
+   - Workflows vary widely among researchers
+   - Users may come up with use cases beyond the original scope
+   - Research leads to new quantities of interest, software needs to be
+     extensible to support these new properties.
+   - New algorithms for computing a property emerge. Need to be able to use
+     these algorithms throughout the code.
+
+#. Complex nature of scientific research
+
+   - Scientific simulations of real world phenomenon have many pieces.
+   - Domains are often hard to grasp for non-experts.
+   - Often multi-disciplinary.
+
+#. Need for rapid prototyping
+
+   - Design space for most scientific algorithms is huge. Need to be able to
+     quickly scan this space.
+   - Python is at present the *de facto* language of choice for rapid
+     prototyping
+
+#. Decentralized scientific software development
+
+   - Developers are typically spread out across the world. Makes synchronizing
+     difficult.
+   - Entire range of software engineering capabilities. Quality of contributions
+     and software products varies widely.
+   - Need to protect unpublished research
```

Comment:
I am not totally sure why this is in the decentralized section. It might fit better in the scientific motivation section?

Reply:
I added a clarification sentence.